lib/src/controllers/pod_player_controller.dart

```diff
@@ -35,17 +39,24 @@ class PodPlayerController {
       );
   }
 
-  /// Initialsing video player
+  /// Initializes the video player.
+  ///
+  /// If the provided video cannot be loaded, an exception could be thrown.
   Future<void> initialise() async {
     if (!_isCtrInitialised) {
       _init();
     }
     WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
-      if (!_isCtrInitialised) {
-        await _ctr.videoInit();
-        podLog('$getTag Pod player Initialized');
-      } else {
-        podLog('$getTag Pod Player Controller Already Initialized');
+      try {
+        if (!_isCtrInitialised) {
+          await _ctr.videoInit();
+          podLog('$getTag Pod player Initialized');
+        } else {
+          podLog('$getTag Pod Player Controller Already Initialized');
+        }
+      } catch (error) {
+        podLog('$getTag Pod Player Controller failed to initialize');
+        _initializationError = error;
```

Comment:
Is it necessary to have `_initializationError`  variable....?
instead, we can rethrow the error directly right?
```dart
  } catch (error) {
        podLog('$getTag Pod Player Controller failed to initialize');
        rethrow;
      }
```

Reply:
Please verify this so I can merge changes

Reply:
Actually, this is the main problem... The rethrow doesn't throws the error to the outside caller, cause it's inside a addPostFrameCallback... 
Moving the responsibility of throwing the error to the `_checkAndWaitTillInitialized` solves this problem

Reply:
Hoo, I was not aware of that Thank you ❤️